### ISSUE TITLE
Create RAD form on Website

### ISSUE THREAD

**Issue by patphongs** [18F/fec-cms#612]
A RAD form needs to be developed using existing OpenFEC styling and be able to submit information to the ServiceNow API. 

- [x] Establish method(s) on authentication between OpenFEC API and ServiceNow API
- [x] Develop form given RAD's specifications
- [x] Test form to ensure proper exchange of information
- [x] Validate final functionality and fix any bugs

**Comment by noahmanger** [comment]
For further background: The basic idea here is that RAD is contracting with ServiceNow to have a public-facing contact form that committees and anyone else can use to submit questions to RAD from the website.

There will be a UX task in here as part of the compliance content work to figure out how we want to send people to this form and where it lives. 

For now, @patphongs, John and I just got off the phone with folks at ServiceNow to discuss the technical details and here's what we know:

- We will need to make a form on a page that gathers:
  - Name
  - Committee ID (could use the typeahead)
  - Email address
  - Their question
  - The category the question falls into (from a select list)
- They will provide a REST API that we can send the data to

This shouldn't be hard, from a technical standpoint. They're still finishing things up on their end, and we discussed us using Sprint 2 to put together a prototype of this form to just make sure that the API connection works correctly.

**Comment by jenniferthibault** [comment]
@noahmanger do you think it's better to do the technical/API prototype before/concurrently/after an interface prototype? 

Initial design questions around this:
- Where does this live? How/where should users find this?
- What expectations can we provide users after they've submitted the contact form?

**Comment by patphongs** [comment]
@noahmanger @jenniferthibault Here's the user story from RAD for this contact page.

1) Customer can access the submission form page from www.fec.gov/rad
2) Page title "Submit a Question to the Reports Analysis Division"
3) Disclaimer text (RAD will provide) to indicate to user that this submission will not constitute filing, etc.
"Submit a question for the Reports Analysis Division:
Reports Analysis Division staff can help answer your filing and reporting questions.  If you are an authorized representative of a committee or another entity registered with the FEC, please complete the fields below to submit a question to the Reports Analysis Division.  
For immediate answers to commonly asked questions, please consult our FAQs [link].
Disclaimer:
Please note, submission of a question through this portal does not constitute a response to a Request for Additional Information (RFAI), a Notice of Failure to File, or any other written communication from the FEC and therefore, will not be accepted.  To respond to a written communication from the FEC, please follow the instructions provided in the communication.  If your inquiry is better suited for resolution over the phone, please contact us at (800) 424-9530 for assistance (press 5 for the Reports Analysis Division)."
4) Required Fields
      - Committee ID (lookup field from the Data.gov API)
            - Popup/help info link for how to look up the committee ID/name if the user does not know it
      - Committee Name (lookup field from the Data.gov API)
      - Contact First Name (free-text field)
      - Contact Last Name (free-text field)
      - Contact Email (free-text field)
      - Category (predefined list with option for "Other)"
             - If "Other" option is selected, display an "Other Reason" free-text field
      - Committee Member Certification (checkbox with verbiage that will be provided by RAD group)
      - Question(free-text)
5) Optional Field
        - Contact Title/Position (free-text field)
6) The committee certification would pop up after the form is submitted and the committee/filer would have to agree before the ticket can be submitted.  The certification states:
"By submitting this inquiry, I certify that I am an authorized representative of the indicated committee or filing entity.  
Please be advised that knowingly and willfully making any materially false, fictitious, or fraudulent statement or representation to a federal government agency, including the Federal Election Commission, is punishable under the provisions of 18 U.S.C. § 1001. The Commission may report apparent violations to the appropriate law enforcement authorities. 52 U.S.C. § 30107(a)(9)."

The pop up is needed to deter frivolous inquiries that are not from an authorized representative from the committee or filing entity.

**Comment by AmyKort** [comment]
Some options for where this might live on the beta site for now:

In the footer of these pages? https://beta.fec.gov/registration-and-reporting/

Here? https://beta.fec.gov/contact-us/

Both places?

Will we be able to easily find a new home for this function as we continue to add content to the beta site?

**Comment by emileighoutlaw** [comment]
Based on the content strategy decisions we've made so far, I think it makes sense for this to live with our registration and reporting pages. The mental model I'm imagining is:

1. Filer goes to FEC.gov looking for the answer to a reporting question.
2. Filer looks for that answer in the "reporting" section of the website.
3. When answer isn't easily accessible, the filer submits an inquiry to RAD.

My only hesitation to putting a form like this directly on the contact page is that I remember from way back in an early design studio that we wanted people to first look for answers on the website rather than first reaching for contact information and calling the FEC. 

So that's my early gut answer to _where_ this should live, @jenniferthibault. But I share your questions about how users find/interact with it and what expectations we can provide after they hit the submission button.

**Comment by AmyKort** [comment]
I agree with @emileighoutlaw 

Thanks for putting that so succinctly.

**Comment by noahmanger** [comment]
@jenniferthibault to answer your question (very lately), I think we can work on the technical prototype in parallel...we basically just need to make a proof of concept form that can hook up to ServiceNow. 

As far as where it lives and how people get to it, those are questions we should answer as we're designing the "Help with reporting and compliance" section this next sprint. (FWIW, I think the form should live on its own page that exists within that section (as it will only be for that audience), and I think it makes sense to have the link to it be always present in a consistent location throughout the section.)

But yeah, we'll also need a separate task for the actual interaction design of this form, especially given the lengthy copy / disclaimers.

**Comment by AmyKort** [comment]
Thinking ahead: My understanding is that we'll eventually have a similar feature for folks to contact information division, press, congressional affairs. I don't know whether that should factor into our thinking. This form is only for RAD, but the contact feature will span more offices.

**Comment by jwchumley** [comment]
I got a question about this today from RAD wanting to know when it will be done. Maybe we should consider it for the next sprint?

**Comment by noahmanger** [comment]
Yep, we can definitely prioritize for next sprint. We'll need to tackle UX design as well as technical stuff. Have we received details about how our form is supposed to submit to that service?

**Comment by noahmanger** [comment]
Ok, I think I got this pretty far. Just a few remaining things to figure out, which I discuss in #755. Handing off to @patphongs and @johnnyporkchops to pick up while I'm out.

**Comment by noahmanger** [comment]
Resolved by #755. This will be deploying to the develop environment soon so @llienfec can take a look.
